content-tool/tool/src/java/org/sakaiproject/content/tool/FilePickerAction.java

```diff
@@ -722,8 +722,20 @@ else if(contentService.isDropboxMaintainer())
 				{
 					items = filterList(items, filter);
 				}
+				//Check if the ListItem in 'items' matches with the attach_item in the new_items list , if yes then it should not have option to be selected.
+				if(items != null && !items.isEmpty() && !new_items.isEmpty()){
```

Comment:
`item.convert2list()` should never return `null` I think (correct?) and `filterList` should never return `null`,  So I'd just drop this `if` completely as is new_item is empty the loop won't run anyway. It makes the code more readable and putting null checks in makes other people worry when reading it what situation you were attempting to cope for.